src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -226,36 +230,66 @@ public void amendHearing(CaseData caseData, String caseTypeId) {
         }
     }
 
+    public void midEventAmendHearing(CaseData caseData, List<String> errors) {
+        if (!CollectionUtils.isEmpty(caseData.getHearingCollection())) {
+            for (HearingTypeItem hearingTypeItem : caseData.getHearingCollection()) {
+                if (!CollectionUtils.isEmpty(hearingTypeItem.getValue().getHearingDateCollection())) {
+                    for (DateListedTypeItem dateListedTypeItem
+                            : hearingTypeItem.getValue().getHearingDateCollection()) {
+                        addError(dateListedTypeItem, errors);
+                    }
+                }
+            }
+        }
+    }
+
+    private void addError(DateListedTypeItem dateListedTypeItem, List<String> errors) {
+        var date = LocalDateTime.parse(
+                dateListedTypeItem.getValue().getListedDate(), OLD_DATE_TIME_PATTERN).toLocalDate();
+        if (DayOfWeek.SUNDAY.equals(date.getDayOfWeek())
```

Comment:
You're calling LocalDate.getDayOfWeek twice.

Reply:
refactored, calling once now.